shared/util/file.native.js

```diff
@@ -1,8 +1,11 @@
 // @flow
+import {NativeModules} from 'react-native'
+
+const nativeBridge = NativeModules.KeybaseEngine || NativeModules.ObjcEngine
 
 // TODO
 function tmpFile (suffix: string): string {
-  throw new Error('Unimplemented')
+  return `${nativeBridge.tmpDir}${suffix}`
```

Comment:
Actually, it has everything to do with it! We weren't downloading previews because this code was needed to generate the destination path previews would be downloaded to. I needed to obtain a temp/cache location for each platform and plumb it into here. Sorry I didn't explain this in the PR description.